tutorials/time-freq/plot_ssvep.py

```diff
@@ -0,0 +1,698 @@
+        increment by one to add one frequency bin ON BOTH SIDES
+    noise_skip_neighborfreqs - int
+        set this >=1 if you want to exclude the immediately neighboring
+        frequency bins in noise level calculation
+
+    Returns
+    -------
+    snr - np.array
+        array containing snr for all epochs, channels, frequency bins.
+        NaN for frequencies on the edge, that do not have enoug neighbors on
+        one side to calculate snr
+    """
+    # Construct a kernel that calculates the mean of the neighboring
+    # frequencies
+    averaging_kernel = np.concatenate((
+        np.ones(noise_n_neighborfreqs),
+        np.zeros(2 * noise_skip_neighborfreqs + 1),
+        np.ones(noise_n_neighborfreqs)))
+    averaging_kernel /= averaging_kernel.sum()
+
+    # Calculate the mean of the neighboring frequencies by convolving with the
+    # averaging kernel.
+    mean_noise = np.apply_along_axis(
+        lambda psd_: np.convolve(psd_, averaging_kernel, mode='valid'),
+        axis=-1, arr=psd
+    )
+
+    # The mean is not defined on the edges so we will pad it with nas. The
+    # padding needs to be done for the last dimension only so we set it to
+    # (0, 0) for the other ones.
+    edge_width = noise_n_neighborfreqs + noise_skip_neighborfreqs
+    pad_width = [(0, 0)] * (mean_noise.ndim - 1) + [(edge_width, edge_width)]
+    mean_noise = np.pad(
+        mean_noise, pad_width=pad_width, constant_values=np.nan
+    )
+
+    return psd / mean_noise
+
+
+###############################################################################
+# Now we call the function to compute our SNR spectrum.
+#
+# As described above, we have to define two parameters.
+#
+# * how many noise bins do we want?
+#
+# * do we want to skip the n bins directly next to the target bin?
+#
+#
+# Tweaking these parameters *can* drastically impact the resulting spectrum,
+# but mainly if you choose extremes.
+# E.g. if you'd skip very many neighboring bins, broad band power modulations
+# (such as the alpha peak) should reappear in the SNR spectrum.
+# On the other hand, if you skip none you might miss or smear peaks if the
+# induced power is distributed over two or more frequency bins (e.g. if the
+# stimulation frequency isn't perfectly constant, or you have very narrow
+# bins).
+#
+# Here, we want to compare power at each bin with average power of the
+# **three neighboring bins** (on each side) and **skip one bin** directly next
+# to it.
+#
+
+
+snrs = snr_spectrum(psds, noise_n_neighborfreqs=3,
+                    noise_skip_neighborfreqs=1)
+
+##############################################################################
+# Plot PSD and SNR spectra
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# Now we will plot grand average PSD (in blue) and SNR (in red) ± sd
+# for every frequency bin.
+# PSD is plotted on a log scale.
+#
+
+fig, axes = plt.subplots(2, 1, sharex='all', sharey='none', dpi=300)
```

Comment:
```suggestion
fig, axes = plt.subplots(2, 1, sharex='all', sharey='none', figsize=(8, 5))
```
dpi is too different between displays.